tsdb/head_append.go

```diff
@@ -846,16 +846,17 @@ func (a *headAppender) Commit() (err error) {
 		// number of samples rejected due to: out of bounds: with t < minValidTime (OOO support disabled)
 		floatOOBRejected int
 
-		inOrderMint     int64 = math.MaxInt64
-		inOrderMaxt     int64 = math.MinInt64
-		ooomint         int64 = math.MaxInt64
-		ooomaxt         int64 = math.MinInt64
-		wblSamples      []record.RefSample
-		oooMmapMarkers  map[chunks.HeadSeriesRef]chunks.ChunkDiskMapperRef
-		oooRecords      [][]byte
-		oooCapMax       = a.head.opts.OutOfOrderCapMax.Load()
-		series          *memSeries
-		appendChunkOpts = chunkOpts{
+		inOrderMint         int64 = math.MaxInt64
+		inOrderMaxt         int64 = math.MinInt64
+		ooomint             int64 = math.MaxInt64
+		ooomaxt             int64 = math.MinInt64
+		wblSamples          []record.RefSample
+		oooMmapMarkers      map[chunks.HeadSeriesRef][]chunks.ChunkDiskMapperRef
```

Comment:
Could you explain why there is now a slice of  `chunks.ChunkDiskMapperRef`? I think I need to fully understand this bit because the rest of the modifications depend on it.

Reply:
When we start adding native histogram samples to the OOO head chunk, when the OOO head chunk is encoded (e.g. when mmapping or reading from the head), it's possible for multiple encoded chunks to be created from the single head chunk. Since multiple mmapped chunks can be created, we need to add multiple mmap markers, to be tracked by the new `chunks.ChunkDiskMapperRef` slice.

Multiple encoded chunks is because of two reasons:
- A native histogram chunk can't have a counter reset within it. For the OOO head, since the samples arrive out of order, we can't figure out the counter resets until we want to encode the samples into chunks.
- The samples in the head chunk might be of different types. E.g. someone first sends some float samples, then some native histogram samples. Each sample type goes in a separate chunk.

Reply:
Understood thank you!